### ISSUE TITLE
[Problem]: No sound coming out with PulseAudio

### ISSUE THREAD

**Issue by Gwouigwoui** [mikebrady/shairport-sync#1751]
### What happened?

Hi, my iPhone can connect, but no sound is coming out. There's an alternance of "large positive sync error" and "large negative sync error" in the log. I followed the guide for building with PulseAudio support.

If I try to switch to Alsa, I get the following error:
>          2.199382239 "audio_alsa.c:458" *warning: The output device "hw:MYDAC" is busy and can't be used by Shairport Sync at present.

Any help would be welcome!

### Relevant log output

```shell
0.002094948 "shairport.c:2252" Startup in AirPlay 2 mode, with features 0x405c4a00,0x1c340 on device "b8:27:eb:ee:d1:c8".
         0.000184271 "shairport.c:2291" Version String: "4.3.2-dev-71-g40306cc7-AirPlay2-smi10-OpenSSL-Avahi-ALSA-pa-soxr-sysconfdir:/etc"
         0.000033437 "shairport.c:2310" Command Line: "shairport-sync -v".
         0.000527084 "shairport.c:2346" Log Verbosity is 1.
         0.081246823 "shairport.c:2394" disable_resend_requests is off.
         0.000101093 "shairport.c:2395" diagnostic_drop_packet_fraction is 0.000000. A value of 0.0 means no packets will be dropped deliberately.
         0.000036823 "shairport.c:2399" statistics_requester status is 0.
         0.000027969 "shairport.c:2404" rtsp listening port is 7000.
         0.000028229 "shairport.c:2405" udp base port is 6001.
         0.000026459 "shairport.c:2406" udp port range is 10.
         0.000026302 "shairport.c:2407" player name is "Raspberrypi".
         0.000027135 "shairport.c:2408" backend is "pa".
         0.000026875 "shairport.c:2409" run_this_before_play_begins action is "(null)".
         0.000027292 "shairport.c:2410" run_this_after_play_ends action is "(null)".
         0.000026510 "shairport.c:2411" wait-cmd status is 0.
         0.000026875 "shairport.c:2412" run_this_before_play_begins may return output is 0.
         0.000026771 "shairport.c:2413" run_this_if_an_unfixable_error_is_detected action is "(null)".
         0.000027344 "shairport.c:2415" run_this_before_entering_active_state action is  "(null)".
         0.000033125 "shairport.c:2417" run_this_after_exiting_active_state action is  "(null)".
         0.000028802 "shairport.c:2419" active_state_timeout is  10.000000 seconds.
         0.000031041 "shairport.c:2420" mdns backend "(null)".
         0.000024636 "shairport.c:2422" interpolation setting is "auto".
         0.000024479 "shairport.c:2426" interpolation soxr_delay_threshold is 30000000.
         0.000024167 "shairport.c:2427" resync time is 0.050000 seconds.
         0.000027396 "shairport.c:2428" resync recovery time is 0.100000 seconds.
         0.000027135 "shairport.c:2429" allow a session to be interrupted: 0.
         0.000054583 "shairport.c:2430" busy timeout time is 0.
         0.000028386 "shairport.c:2431" drift tolerance is 0.002000 seconds.
         0.000026198 "shairport.c:2432" password is "(null)".
         0.000047708 "shairport.c:2433" default airplay volume is: -24.000000.
         0.000032708 "shairport.c:2434" high threshold airplay volume is: -16.000000.
         0.000027657 "shairport.c:2436" check for higher-than-threshold volume for new play session is disabled.
         0.000029114 "shairport.c:2442" ignore_volume_control is 0.
         0.000183021 "shairport.c:2446" volume_max_db is not set
         0.000040052 "shairport.c:2447" volume range in dB (zero means use the range specified by the mixer): 0.
         0.000046458 "shairport.c:2449" volume_range_combined_hardware_priority (1 means hardware mixer attenuation is used first) is 0.
         0.000030469 "shairport.c:2453" playback_mode is 0 (0-stereo, 1-mono, 1-reverse_stereo, 2-both_left, 3-both_right).
         0.000041458 "shairport.c:2455" disable_synchronization is 0.
         0.000027032 "shairport.c:2456" use_mmap_if_available is 1.
         0.000026510 "shairport.c:2457" output_format automatic selection is enabled.
         0.000026667 "shairport.c:2461" output_rate automatic selection is enabled.
         0.000026823 "shairport.c:2465" audio backend desired buffer length is 0.350000 seconds.
         0.000028437 "shairport.c:2467" audio_backend_buffer_interpolation_threshold_in_seconds is 0.020000 seconds.
         0.000035000 "shairport.c:2469" audio backend latency offset is 0.000000 seconds.
         0.000028698 "shairport.c:2471" audio backend silence lead-in time is "auto".
         0.000026615 "shairport.c:2475" zeroconf regtype is "_raop._tcp".
         0.000027864 "shairport.c:2476" decoders_supported field is 1.
         0.000027969 "shairport.c:2477" use_apple_decoder is 0.
         0.000026406 "shairport.c:2478" alsa_use_hardware_mute is 0.
         0.000026771 "shairport.c:2482" no special mdns service interface was requested.
         0.000066302 "shairport.c:2485" configuration file name "/etc/shairport-sync.conf" resolves to "/etc/shairport-sync.conf".
         0.000031615 "shairport.c:2515" loudness is 0.
         0.000030937 "shairport.c:2516" loudness reference level is -20.000000
         0.001397865 "shairport.c:2643" NQPTP is online.
         1.503807291 "shairport.c:255" "soxr" interpolation has been chosen.
         2.831690259 "rtsp.c:2902" Connection 2: AP2 PTP connection from XXX to self at YYY.
         0.230469584 "rtsp.c:3292" Connection 2. AP2 Realtime Audio Stream.
         0.025506093 "rtp.c:1738" AP2 Realtime Clock receiver initialised.
         1.796839166 "player.c:2881" Large positive (i.e. late) sync error of 4950 frames (0.112245 seconds), at frame: 3530309485.
         0.300409688 "player.c:2906" Large negative (i.e. early) sync error of -11151 frames (-0.252857 seconds), at frame: 3530322861.
         0.114095416 "player.c:2881" Large positive (i.e. late) sync error of 4552 frames (0.103220 seconds), at frame: 3530327437.
         0.300669167 "player.c:2906" Large negative (i.e. early) sync error of -10462 frames (-0.237234 seconds), at frame: 3530340461.
         0.092253021 "player.c:2881" Large positive (i.e. late) sync error of 3718 frames (0.084308 seconds), at frame: 3530345037.
         0.279798437 "player.c:2906" Large negative (i.e. early) sync error of -10950 frames (-0.248299 seconds), at frame: 3530357357.
```

### System Information.

USB DAC connected to a Raspberry Pi 4 using latest Raspberry Pi OS. Only other use is Plexamp (which uses PulseAudio and works fine).

### Configuration Information.

```shell
>> Display Config Start.

From "uname -a":
 Linux raspberrypi 6.1.0-rpi4-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.54-1+rpt2 (2023-10-05) aarch64 GNU/Linux

From /etc/os-release:
 Debian GNU/Linux 12 (bookworm)

From /sys/firmware/devicetree/base/model:
 Raspberry Pi 3 Model B Plus Rev 1.3

Shairport Sync Version String:
 4.3.2-dev-71-g40306cc7-AirPlay2-smi10-OpenSSL-Avahi-ALSA-pa-soxr-sysconfdir:/etc

Command Line:
 shairport-sync --displayConfig

Configuration File:
 /etc/shairport-sync.conf

Configuration File Settings:
 general :
 {
   output_backend = "pa";
 };
 pw :
 {
 };

>> Display Config End.
>> Goodbye!
```

### PulseAudio or PipeWire installed?

- [X] Check if your system uses a Sound Server.

### How did you install Shairport Sync?

Built from source

### Check previous issues

- [X] Confirm

**Comment by mikebrady** [comment]
Thanks for the post.

The latest Raspberry Pi OS -- Bookworm -- uses PipeWire rather than PulseAudio, so I'd suggest you try out the PipeWire backend. Otherwise, the PulseAudio → PipeWire adapter is used. My slight experience is that it's better to go directly to the PipeWire system.

Here are some statistics from an otherwise-idle Raspberry Pi 4 running Bookworm. Shairport Sync has been compiled for both PipeWire and PulseAudio:
`4.3.2-1-g90c03bf0-AirPlay2-smi10-OpenSSL-Avahi-pa-pw-soxr-sysconfdir:/etc`:

```
$ shairport-sync -v --statistics -o pa
...
...
         0.000920037 "shairport.c:2643" NQPTP is online.
         1.501914962 "shairport.c:255" "soxr" interpolation has been chosen.
        51.101864507 "rtsp.c:2902" Connection 1: AP2 PTP connection from 192.168.50.108:51930 ("Mike Brady’s iPad mini") to self at 192.168.50.43:7000.
         1.490867870 "player.c:2835" Connection 1: Playback started at frame 1314779047 -- AirPlay 2 Buffered.
         7.929998995 "player.c:2654" Sync Error ms | Net Sync PPM | All Sync PPM | Min DAC Queue | Min Buffers | Min Buffer Size
         0.000540963 "player.c:2654"         26.06         -947.0          947.0           14100            46              223k
         8.000754051 "player.c:2654"          2.09         -682.6         1317.1           13646            46             2973k
         8.020002884 "player.c:2654"         -2.36         1022.5         1181.1           13901            46             2721k
         7.995947717 "player.c:2654"         -0.52         -169.9          186.9           13940            46             2460k
         8.044386088 "player.c:2654"         -4.27         1617.3         1657.0           12991            46             2181k
         7.983439477 "player.c:2654"         -0.29         -116.1          257.7           13932            46             1918k
         8.024444402 "player.c:2654"         -0.42          -87.8          195.4           13929            46             1650k
         8.004707699 "player.c:2654"         -0.78          -76.5          195.4           13929            46             1369k
...
```

You can see a good deal of interpolation is going on, which is undesirable.

With the PipeWire backend, results are better:
```
$ shairport-sync -v --statistics -o pw
...
...
         1.602443184 "player.c:2835" Connection 1: Playback started at frame 3606765023 -- AirPlay 2 Buffered.
         7.925851718 "player.c:2654" Sync Error ms | Net Sync PPM | All Sync PPM | Min DAC Queue | Min Buffers | Min Buffer Size
         0.000180926 "player.c:2654"         -0.12            0.0            0.0           14020            46              219k
         7.999035365 "player.c:2654"         -0.54            0.0            0.0           13996            46             3868k
         8.012402773 "player.c:2654"         -0.72            0.0            0.0           13987            46             3809k
         7.999077996 "player.c:2654"         -0.88            0.0            0.0           13985            46             3862k
         7.997332347 "player.c:2654"         -0.96            0.0            0.0           13963            46             3870k
         7.999171828 "player.c:2654"         -1.03            0.0            0.0           13982            46             3863k
         8.031830310 "player.c:2654"         -0.84            0.0            0.0           13986            46             3852k
         7.979276144 "player.c:2654"         -0.58            0.0            0.0           13976            46             3783k
         8.026546939 "player.c:2654"         -0.50            0.0            0.0           14000            46             3718k
         7.988760847 "player.c:2654"         -0.35            0.0            0.0           14003            46             3703k
         8.013036273 "player.c:2654"         -0.44            0.0            0.0           14003            46             3694k
         7.998560514 "player.c:2654"         -0.50            0.0            0.0           13990            46             3747k
         8.004465458 "player.c:2654"         -0.47            0.0            0.0           14000            46             3681k
         8.015250014 "player.c:2654"         -0.42            0.0            0.0           13998            46             3668k
         8.000533699 "player.c:2654"         -0.54            0.0            0.0           13982            46             3662k
         7.996459977 "player.c:2654"         -0.44            0.0            0.0           13996            46             3644k
         8.006005976 "player.c:2654"         -0.34            0.0            0.0           14009            46             3682k
         8.033916940 "player.c:2654"         -0.17            0.0            0.0           14008            46             3618k
         7.982683681 "player.c:2654"         -0.04            2.8            2.8           13493            46             3583k
         8.007303347 "player.c:2654"         -0.14            0.0            0.0           13989            46             3552k
...
```
Just as a matter of interest, your "Configuration Information" reports a Raspberry Pi 3B. Perhaps the amount of interpolation, which is computationally demanding, is causing the problem on the Pi 3?

**Comment by Gwouigwoui** [comment]
Ok, I rebuilt with both PulseAudio and Pipewire backends, and tried that.
I'm getting a different error with Pipewire, but the result is the same: no sound.
`         0.269830156 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.001856614 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.019372292 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000179063 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000069635 "player.c:2906" Large negative (i.e. early) sync error of -10902 frames (-0.247211 seconds), at frame: 4211420607.
         0.089725365 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000385208 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000315677 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.022439323 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000285677 "player.c:2881" Large positive (i.e. late) sync error of 4628 frames (0.104943 seconds), at frame: 4211425183.
         0.255237240 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.028690729 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000196198 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
         0.000126562 "audio_pw.c:354" *warning: Shairport Sync's PipeWire backend can not get timing information from the PipeWire system. Is PipeWire running?
`

Pipewire is running, I think, as this is the output when I run "systemctl --user status pipewire"
`● pipewire.service - PipeWire Multimedia Service
     Loaded: loaded (/usr/lib/systemd/user/pipewire.service; enabled; preset: enabled)
     Active: active (running) since Sat 2023-10-28 15:29:34 EDT; 11min ago
TriggeredBy: ● pipewire.socket
   Main PID: 832 (pipewire)
      Tasks: 3 (limit: 758)
        CPU: 477ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire.service
             └─832 /usr/bin/pipewire

oct. 28 15:36:27 raspberrypi pipewire[832]: pw.node: (alsa_output.usb-MICROMEGA_MICROMEGA_MYDAC_0000-00.analog-stereo-32) suspended -> error (Start error: Périphérique ou ressource occupé)
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.audioadapter: params Spa:Enum:ParamId:EnumFormat: 0:0 (follower format) Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.audioadapter: params Spa:Enum:ParamId:EnumFormat: 0:0 (follower format) Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.alsa: 'front:2': playback open failed: Périphérique ou ressource occupé
oct. 28 15:36:28 raspberrypi pipewire[832]: spa.audioadapter: params Spa:Enum:ParamId:EnumFormat: 0:0 (follower format) Périphérique ou ressource occupé
`

And "sudo lsof -n /dev/snd/*" returns
`lsof: WARNING: can't stat() fuse.gvfsd-fuse file system /run/user/1000/gvfs
      Output information may be incomplete.
COMMAND   PID      USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
pipewire  832 guillaume   52u   CHR  116,1      0t0  429 /dev/snd/seq
pipewire  832 guillaume   53u   CHR  116,1      0t0  429 /dev/snd/seq
wireplumb 834 guillaume   42u   CHR 116,32      0t0  589 /dev/snd/controlC1
wireplumb 834 guillaume   43u   CHR 116,64      0t0  630 /dev/snd/controlC2
`

**Comment by mikebrady** [comment]
Thanks for the update. Is there anything useful on the [PulseAudioAndPipeWire](https://github.com/mikebrady/shairport-sync/blob/master/ADVANCED%20TOPICS/PulseAudioAndPipeWire.md) page?

**Comment by Gwouigwoui** [comment]
Unfortunately no. Here's the output of sps-alsa-explore:
```
> Device Full Name:    "hw:Headphones"
  Short Name:          "hw:1"
  This device seems suitable for use with Shairport Sync.
  Possible mixers:     "PCM",0                 Range: 106.38 dB
  The following rate and format would be chosen by Shairport Sync in "auto" mode:
     Rate              Format
     44100             S16_LE

> Device Full Name:    "hw:MYDAC"
  Short Name:          "hw:2"
  This device seems suitable for use with Shairport Sync.
  The following rate and format would be chosen by Shairport Sync in "auto" mode:
     Rate              Format
     44100             S32_LE
```

**Comment by mikebrady** [comment]
Thanks again. Here are those results for this 8GB Pi 4, which has a standard desktop image, is fully up-to-date, with Shairport Sync, NQPTP and sps-alsa-explore built on it and installed. It also has a Pimoroni PHAT DAC, using the `dtoverlay=hifiberry-dac` overlay, installed. Linux version:
```
$ uname -a
Linux RaspberryPi4B 6.1.0-rpi4-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.54-1+rpt2 (2023-10-05) aarch64 GNU/Linux
```
PipeWire User Status:
```
$ systemctl --user status pipewire
● pipewire.service - PipeWire Multimedia Service
     Loaded: loaded (/usr/lib/systemd/user/pipewire.service; enabled; preset: enabled)
     Active: active (running) since Sun 2023-10-29 08:58:05 GMT; 4min 58s ago
TriggeredBy: ● pipewire.socket
   Main PID: 1371 (pipewire)
      Tasks: 3 (limit: 8734)
        CPU: 128ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire.service
             └─1371 /usr/bin/pipewire

Oct 29 08:58:05 RaspberryPi4B systemd[900]: Started pipewire.service - PipeWire Multimedia Service.
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video14' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video15' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video21' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video22' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video14' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video15' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video21' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
Oct 29 08:58:06 RaspberryPi4B pipewire[1371]: spa.v4l2: '/dev/video22' VIDIOC_ENUM_FRAMEINTERVALS: Inappropriate ioctl for device
```
SPS Alsa Explore:
```
$ ./sps-alsa-explore 
> Device Full Name:    "hw:vc4hdmi0"
  Short Name:          "hw:0"
  Shairport Sync can not use this device.

> Device Full Name:    "hw:Headphones"
  Short Name:          "hw:2"
  This device seems suitable for use with Shairport Sync.
  Possible mixers:     "PCM",0                 Range: 106.38 dB
  The following rate and format would be chosen by Shairport Sync in "auto" mode:
     Rate              Format
     44100             S16_LE

> Device Full Name:    "hw:sndrpihifiberry"
  Short Name:          "hw:3"
  This device seems suitable for use with Shairport Sync.
  The following rate and format would be chosen by Shairport Sync in "auto" mode:
     Rate              Format
     44100             S32_LE

```
Finally:
```
# lsof -n /dev/snd/*
lsof: WARNING: can't stat() fuse.gvfsd-fuse file system /run/user/1000/gvfs
      Output information may be incomplete.
lsof: WARNING: can't stat() fuse.portal file system /run/user/1000/doc
      Output information may be incomplete.
COMMAND    PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
pipewire  1371   pi   49u   CHR  116,1      0t0  401 /dev/snd/seq
pipewire  1371   pi   50u   CHR  116,1      0t0  401 /dev/snd/seq
wireplumb 1375   pi   41u   CHR 116,64      0t0  561 /dev/snd/controlC2
wireplumb 1375   pi   42u   CHR 116,96      0t0  580 /dev/snd/controlC3
wireplumb 1375   pi   44u   CHR  116,0      0t0  329 /dev/snd/controlC0
```

The system is outputting audio through the `Headphones` device.

**Comment by Gwouigwoui** [comment]
Ok, I did a full reinstall and now everything's working fine with Pipewire. One last step I can't figure out is how to get shairport to start at boot. I tried [this ](https://github.com/mikebrady/shairport-sync/issues/1749#issuecomment-1778622232), to no avail. I'm only connecting through ssh, if that makes a difference.

**Comment by mikebrady** [comment]
It's good that you're making progress 🙂.

If you're using PipeWire, your user must log in through the desktop GUI before the PipeWire service becomes available. This makes it unsuitable, IMHO, for unattended operation, such as for a stand-alone player. (BTW, I have not tried to see what happens if the desktop user is automatically logged in at startup.) 

This whole issue is dealt with in more detail on the [PulseAudioAndPipeWire](https://github.com/mikebrady/shairport-sync/blob/master/ADVANCED%20TOPICS/PulseAudioAndPipeWire.md) page.

**Comment by Gwouigwoui** [comment]
That's curious: I'm using a headless Raspberry, and didn't set-up autologin to GUI in raspi-config, and yet PipeWire is running (the process shows up in 'ps -e'). Lucky me I guess! Now I just need to figure out how to create the user service startup script.

**Comment by Gwouigwoui** [comment]
Oh yes, it's the standard version, as the Lite version wouldn't work with [this other script](https://github.com/fdanis-oss/pw_wp_bluetooth_rpi_speaker) which turns the Pi into a Bluetooth speaker (this one requires PipeWire).

Anyway, for anyone stumbling on this issue, here's what I did:
- Install Raspberry Pi OS 12 64bit (non-Lite)
- Install Shairport-sync following the guide here to the letter. Turned out the script starts as a system service
- Install the bluetooth speaker script referenced above, following it to the letter
- Install Plexamp using [this guide](https://howtohifi.com/install-headless-plexamp-endpoint-home-network-raspberry-pi/)

Now I have three different means of playing music on the Pi: Airplay, Bluetooth, and Plexamp! I tried playing two sources at the same time, Airplay and Bluetooth, and I could hear them both, as I was expecting.

This non-issue can be closed!
